api/k8s/v1/k8s.proto

```diff
@@ -166,3 +174,39 @@ message ResizeHPARequest {
 
 message ResizeHPAResponse {
 }
+
+message Deployment {
+  option (clutch.api.v1.id).patterns = {
+    type_url : "clutch.k8s.v1.Deployment",
+    pattern : "{cluster}/{namespace}/{name}"
+  };
+
+  string cluster = 1;
+  string namespace = 2;
+  string name = 3;
+
+  map<string, string> labels = 4;
+  map<string, string> annotations = 5;
+}
+
+message UpdateDeploymentRequest {
+  option (clutch.api.v1.id).patterns = {
+    type_url : "clutch.k8s.v1.Deployment",
+    pattern : "{cluster}/{namespace}/{name}"
+  };
+
+  string clientset = 1 [ (validate.rules).string = {min_bytes : 1} ];
+  string cluster = 2 [ (validate.rules).string = {min_bytes : 1} ];
+  string namespace = 3 [ (validate.rules).string = {min_bytes : 1} ];
+  string name = 4 [ (validate.rules).string = {min_bytes : 1} ];
+
+  message Fields {
+    map<string, string> labels = 1 [ (validate.rules).map.keys.string = {min_bytes : 1} ];
+    map<string, string> annotations = 2 [ (validate.rules).map.keys.string = {min_bytes : 1} ];
+  }
```

Comment:
can you document the update semantics we discussed earlier?

also guessing it's not possible to delete a label in this API?

Reply:
Yeah sure let me add some documentation to the proto. Currently no there is not a way to delete labels or annotations in this implementation, if we needed that functionality at some point we could add a new message say `message DeleteFields | RemoveFields`.